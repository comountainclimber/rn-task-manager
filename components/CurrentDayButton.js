import React from "react";
import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import moment from "moment";
import PropTypes from "prop-types";

import ChevronDown from "../assets/images/chevron-down.svg";
import ChevronRight from "../assets/images/chevron-right.svg";

export default function CurrentDayButton({
  handlePress,
  selectedDate,
  buttonPressed
}) {
  const day = moment(selectedDate, "YYYY-MM-DD HH:mm:ss").format("dddd");
  const monthAndDay = moment(selectedDate, "YYYY-MM-DD HH:mm:ss").format(
    "MMMM D"
  );

  return (
    <TouchableOpacity onPress={handlePress} style={styles.currentDayContainer}>
      <View style={styles.innerCurrentDayContainer}>
        <Text style={styles.day}>{day}</Text>
        <Text style={styles.monthAndDay}>{monthAndDay}</Text>
      </View>
      {buttonPressed ? <ChevronDown /> : <ChevronRight />}
    </TouchableOpacity>
  );
}

CurrentDayButton.propTypes = {
  buttonPressed: PropTypes.bool.isRequired,
  handlePress: PropTypes.func.isRequired,
  selectedDate: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
  currentDayContainer: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 24
  },
  innerCurrentDayContainer: {
    display: "flex",
    flexDirection: "column",
    marginRight: 12
  },
  day: {
    fontSize: 24,
    fontWeight: "bold",
    width: "100%",
    textAlign: "left"
  },
  monthAndDay: {
    fontSize: 16,
    width: "100%",
    textAlign: "left",

    color: "grey"
  }
});
