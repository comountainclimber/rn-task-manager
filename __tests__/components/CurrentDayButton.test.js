import React from "react";
import renderer from "react-test-renderer";

import CurrentDayButton from "../../components/CurrentDayButton";

const mock = () => undefined;

it("it renders without crashing", () => {
  const tree = renderer
    .create(
      <CurrentDayButton
        buttonPressed
        handlePress={mock}
        selectedDate="10-10-2020"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
