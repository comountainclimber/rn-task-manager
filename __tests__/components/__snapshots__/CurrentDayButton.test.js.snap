// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 24,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginRight": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Nunito-Bold",
          "fontSize": 24,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      Wednesday
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "Nunito-Regular",
          "fontSize": 16,
          "marginTop": -2,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      October 20
    </Text>
  </View>
  <test-file-stub />
</View>
`;
